### PR TITLE
loader: don't print ERROR log, clean dump file when all finished

### LINKED ISSUE #1006: error log when re-init checkpoint for load unit
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

2. What did you expect to see?

3. What did you see instead?

    ```
    [2020/09/08 13:38:21.619 +08:00] [ERROR] [baseconn.go:182] ["execute statement failed"] [task=dm-bi-4304-2] [unit=load] [query="CREATE TABLE `lbs_poi_price` (`poi_id` varchar(64) NOT NULL,`price` double(11,8) NOT NULL,PRIMARY KEY (`poi_id`)) ENGINE=MyISAM DEFAULT CHARSET=utf8 COMMENT='POI-PRICE';"] [argument="[]"] [error="Error 1050: Table 'lbs.lbs_poi_price' already exists"]
    [2020/09/08 13:38:21.620 +08:00] [ERROR] [db.go:176] ["execute statements failed after retry"] [task=dm-bi-4304-2] [unit=load] [queries="[USE `lbs`; CREATE TABLE `lbs_poi_price` (`poi_id` varchar(64) NOT NULL,`price` double(11,8) NOT NULL,PRIMARY KEY (`poi_id`)) ENGINE=MyISAM DEFAULT CHARSET=utf8 COMMENT='POI-PRICE';]"] [arguments="[]"] [error="[code=10006:class=database:scope=not-set:level=high], Message: execute statement failed: CREATE TABLE `lbs_poi_price` (`poi_id` varchar(64) NOT NULL,`price` double(11,8) NOT NULL,PRIMARY KEY (`poi_id`)) ENGINE=MyISAM DEFAULT CHARSET=utf8 COMMENT='POI-PRICE';, RawCause: Error 1050: Table 'lbs.lbs_poi_price' already exists"]
    [2020/09/08 13:38:21.620 +08:00] [INFO] [loader.go:1019] ["table already exists, skip it"] [task=dm-bi-4304-2] [unit=load] ["table schema file"=./dumped_data.dm-bi-4304-2/lbs.lbs_poi_price-schema.sql]
    ```

    ```
    [2020/09/08 12:33:27.717 +08:00] [ERROR] [baseconn.go:182] ["execute statement failed"] [task=dm-bi-4304-2] [unit=load] ["worker ID"=11] [query="INSERT INTO `dm_meta`.`dm-bi-4304-2_loader_checkpoint` (`id`, `filename`, `cp_schema`, `cp_table`, `offset`, `end_pos`) VALUES(?,?,?,?,?,?)"] [argument="[4304-replica-02 lbs.lbs_monitor_action.5.sql lbs lbs_monitor_action 0 97364896]"] [error="Error 1062: Duplicate entry '4304-replica-02-lbs.lbs_monitor_action.5.sql' for key 'uk_id_f'"]
    [2020/09/08 12:33:27.717 +08:00] [ERROR] [db.go:176] ["execute statements failed after retry"] [task=dm-bi-4304-2] [unit=load] ["worker ID"=11] [queries="[INSERT INTO `dm_meta`.`dm-bi-4304-2_loader_checkpoint` (`id`, `filename`, `cp_schema`, `cp_table`, `offset`, `end_pos`) VALUES(?,?,?,?,?,?)]"] [arguments="[[4304-replica-02 lbs.lbs_monitor_action.5.sql lbs lbs_monitor_action 0 97364896]]"] [error="[code=10006:class=database:scope=not-set:level=high], Message: execute statement failed: INSERT INTO `dm_meta`.`dm-bi-4304-2_loader_checkpoint` (`id`, `filename`, `cp_schema`, `cp_table`, `offset`, `end_pos`) VALUES(?,?,?,?,?,?), RawCause: Error 1062: Duplicate entry '4304-replica-02-lbs.lbs_monitor_action.5.sql' for key 'uk_id_f'"]
    [2020/09/08 12:33:27.717 +08:00] [INFO] [checkpoint.go:307] ["checkpoint record already exists, skip it."] [task=dm-bi-4304-2] [unit=load] [component="remote checkpoint"] [id=4304-replica-02] [filename=lbs.lbs_monitor_action.5.sql]
    ```

    if checkpoint for the table `already exists`, we can `skip it`, but we log some `ERROR` and it's misleading.

4. Versions of the cluster

    - DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):

        ```
        v2.0.0-rc
        ```

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
close #1006

### What is changed and how it works?
- **keep checkpoint in memory updated with DB**
- don't execute structure SQL when found db/table in checkpoint
- don't repeatly initiate checkpoint record
- only clean dump SQL files when all finished, to keep progress when resuming

### Check List 

Tests 

 - Unit test
 - Manually check log contains "skip creating it"

Code changes

 - Has exported function/method change
 - Has interface methods change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: loader/checkpoint.go [hunk 1]
```diff
@@ -44,6 +44,9 @@ type CheckPoint interface {
 	// GetAllRestoringFileInfo return all restoring files position
 	GetAllRestoringFileInfo() map[string][]int64
 
+	// IsTableCreated checks if db / table was not created. set `table` to "" when check db
```
Comment (csuzhangxc) [inline loader/checkpoint.go:47]:
```suggestion
	// IsTableCreated checks if db / table was created. set `table` to "" when check db
```

File: loader/loader.go [hunk 2]
```diff
@@ -1295,18 +1273,16 @@ func (l *Loader) cleanDumpFiles() {
...
+		// leave metadata file, only delete sql files
+		files := CollectDirFiles(l.cfg.Dir)
+		var lastErr error
+		for f := range files {
+			if strings.HasSuffix(f, ".sql") {
+				lastErr = os.Remove(f)
 			}
 		}
+		if lastErr != nil {
+			l.logCtx.L().Warn("error when remove loaded dump sql files", zap.String("data folder", l.cfg.Dir), zap.Errors("last error", []error{lastErr}))
```
Comment (csuzhangxc) [inline loader/loader.go:1285]: why do you use `zap.Errors` with only one error?

Comment (lance6716) [reply loader/loader.go]: https://github.com/pingcap/dm/pull/1027/commits/3a629dd9115857c6d47733a573342e7f0f6fd306

File: loader/checkpoint.go (hunk 1 above)
Comment (lance6716) [reply loader/checkpoint.go]: https://github.com/pingcap/dm/pull/1027/commits/3a629dd9115857c6d47733a573342e7f0f6fd306

CI fails, still WIP

File: loader/checkpoint.go [hunk 3]
```diff
@@ -346,6 +381,16 @@ func (cp *RemoteCheckPoint) GenSQL(filename string, offset int64) string {
 	return sql
 }
 
+// UpdateOffset implements CheckPoint.UpdateOffset
+func (cp *RemoteCheckPoint) UpdateOffset(filename string, offset int64) {
+	fields := strings.Split(filename, ".")
```
Comment (GMHDBJD) [inline loader/checkpoint.go:386]: How about use job.schema, job.table instead?

Comment (lance6716) [reply loader/checkpoint.go:386]: I've thinked of that, and found `job.schema` is target schema, here we need source schema. Bad variable name or lack comments 😂 

Hope we could replace it to lightning soon

File: loader/loader.go [hunk 4]
```diff
@@ -262,22 +250,28 @@ func (w *Worker) dispatchSQL(ctx context.Context, file string, offset int64, tab
...
+	// file was not found in checkpoint
+	if offset == uninitializedOffset {
+		offset = 0
 
-	baseFile := filepath.Base(file)
-	err = w.checkPoint.Init(w.tctx.WithContext(ctx), baseFile, finfo.Size())
-	if err != nil {
-		w.tctx.L().Error("fail to initial checkpoint", zap.String("data file", file), zap.Int64("offset", offset), log.ShortError(err))
-		return err
+		finfo, err := f.Stat()
```
Comment (GMHDBJD) [inline loader/loader.go:265]: shadows declaration here

**Review by GMHDBJD — COMMENTED** [review]
LGTM

File: loader/checkpoint.go [hunk 5]
```diff
@@ -346,6 +381,16 @@ func (cp *RemoteCheckPoint) GenSQL(filename string, offset int64) string {
...
+	if len(fields) != 2 && len(fields) != 3 {
+		cp.logCtx.L().Error("can't get db and table from filename in checkpoint UpdateOffset", zap.String("filename", filename))
+	}
+	db, table := fields[0], fields[1]
+	cp.restoringFiles[db][table][filename][0] = offset
```
Comment (csuzhangxc) [inline loader/checkpoint.go:391]: No mutex needed because we only modify `cp.restoringFiles[db][table][filename]` in one goroutine?

**Review by csuzhangxc — APPROVED** [review]
LGTM

File: loader/checkpoint.go (hunk 5 above)
Comment (lance6716) [reply loader/checkpoint.go:391]: structure of map is initiated in `(*RemoteCheckPoint).Init`, thus it won't cause rehash and allocation, won't access invalid memory. same job of same SQL file are processed in one goroutine

**Pull request merged**